src/index.js

```diff
@@ -243,12 +243,16 @@ class Dropzone extends React.Component {
 
   onClick(evt) {
     const { onClick, disableClick } = this.props
-    if (!disableClick) {
-      evt.stopPropagation()
 
-      if (onClick) {
-        onClick.call(this, evt)
-      }
+    // if onClick prop is given, run it first
+    if (onClick) {
+      onClick.call(this, evt)
+    }
+
+    // if disableClick is not set and the event hasn't been default prefented within
+    // the onClick listener, open the file dialog
+    if (!disableClick && !evt.isDefaultPrevented()) {
```

Comment:
Could you add a test for this change?

Reply:
Hi @rolandjitsu, I'm checking `typings/tests` to learn how to add a test, but I do know where the "actions" or "asserts" are written. May I have a short explanation please?

Reply:
@ibc actually the tests are right alongside the source files. So you should be looking at `src/index.spec.js`. I think there are plenty of tests there to illustrate how everything works.

And from the root of the project, after you installed the npm packages, just run `yarn test` or `jest --watch`.

Reply:
ok, thanks, will try.

Reply:
Here you have the requested tests :)

Reply:
Let's also make sure that coverage doesn't drop. If you run `yarn test`/`npm test`, you should see which lines are not covered (ignore two lines: one in utils#L54 and `setTimeout(this.open.bind(this), 0)` in the main file).